### PR TITLE
Added new features for battery module cards

### PR DESCRIPTION
**Pull request by amalia94**

### CODE REVIEW

File: src/front_end/carsdata/app/batteryModule/page.tsx [hunk 1]
```diff
@@ -1,12 +1,16 @@
 import React from 'react'
-import BatteryPercentage from '../components/animations/batteryPercentage.js'
-import BatteryVoltage from '../components/animations/batteryVoltage.js'
-import BatteryStateOfCharge from '../components/animations/batteryState.js'
-import LifeCycle from '../components/animations/lifeCycle.js'
-import BatteryFully from '../components/animations/batteryFully.js'
-import SerialNumber from '../components/animations/serialNumber.js'
-import Other from '../components/animations/other.js'
-import Temperature from '../components/animations/temperature.js'
+import PercentageCard from '../components/batteryComponents/PercentageCard'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/batteryModule/page.tsx:2]: all imports should have ; at the end of row

Comment (Elisei-26) [inline src/front_end/carsdata/app/batteryModule/page.tsx:2]: all imports should have ; at the end of row

File: src/front_end/carsdata/app/components/batteryComponents/BatteryStateCard.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+'use client'
+import React, { useState } from 'react'
+import BatteryStateOfCharge from '../animations/batteryState'
+import Modal from '../sharedComponents/ModalString';
+
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/BatteryStateCard.tsx:5]: ; at the end of import import rows and add the .js extension in animation path

File: src/front_end/carsdata/app/components/batteryComponents/BatteryStateCard.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const BatteryStateCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.BatteryStateOfCharge);
+      const handleInputChange = (value: any, id: string) => {
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/BatteryStateCard.tsx:9]: delete the id parameter, it is not used

File: src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+'use client'
+import React, { useState } from 'react'
+import ChargingTime from '../animations/charging_time'
+import Modal from '../sharedComponents/ModalString'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx:4]: ; at the end of import rows, .js estension for animation path, keep the camel case pattern for animations file name

File: src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+const ChargingTimeCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.chargingTime);
+      const handleInputChange = (value: any, id: string) => {
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx:8]: delete id parameter

File: src/front_end/carsdata/app/components/batteryComponents/DeviceConsumption.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+'use client'
+import React, { useState } from 'react'
+import DeviceConsumption from '../animations/device_consumption'
+import ModalDigit from '../sharedComponents/ModalDigit';
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/DeviceConsumption.tsx:4]: ; at the end of import rows, .js estension for animation path, keep the camel case pattern for animations file name

File: src/front_end/carsdata/app/components/batteryComponents/FullyChargedCard.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+'use client'
+import React, { useState } from 'react'
+import BatteryFully from '../animations/batteryFully'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/FullyChargedCard.tsx:3]: ; at the end of import rows, .js estension for animation path

File: src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
+'use client'
+import React, { useState } from 'react'
+import LifeCycle from '../animations/lifeCycle'
+import Modal from '../sharedComponents/ModalString'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx:4]: ; at the end of import rows, .js estension for animation path

File: src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+const LifeCycleCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.life_cycle);
+      const handleInputChange = (value: any, id: string) => {
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx:8]: delete id parameter

File: src/front_end/carsdata/app/components/batteryComponents/PercentageCard.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
+'use client'
+import React, { useState } from 'react'
+import BatteryPercentage from '../animations/batteryPercentage.js'
+import ModalString from '../sharedComponents/ModalString'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/PercentageCard.tsx:4]: ; at the end of import rows

File: src/front_end/carsdata/app/components/batteryComponents/RangeBatteryCard.tsx [hunk 11]
```diff
@@ -0,0 +1,31 @@
+'use client'
+import React, { useState } from 'react'
+import Other from '../animations/other'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/RangeBatteryCard.tsx:3]: ; at the end of import rows, .js estension for animation path

File: src/front_end/carsdata/app/components/batteryComponents/SerialNumberCard.tsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
+'use client'
+import React, { useState } from 'react'
+import SerialNumber from '../animations/serialNumber'
+import Modal from '../sharedComponents/ModalString';
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/SerialNumberCard.tsx:4]: ; at the end of import rows, .js estension for animation path

File: src/front_end/carsdata/app/components/batteryComponents/SerialNumberCard.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+const SerialNumberCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.serialNumber);
+      const handleInputChange = (value: any, id: string) => {
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/SerialNumberCard.tsx:8]: delete  id parameter

File: src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx [hunk 14]
```diff
@@ -0,0 +1,33 @@
+'use client'
+import React, { useState } from 'react'
+import Temperature from '../animations/temperature'
+import Modal from '../sharedComponents/ModalString'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx:4]: ; at the end of import rows, .js estension for animation path

File: src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+
+const TemperatureCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.temperature);
+      const handleInputChange = (value: any, id: string) => {
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx:8]: delete id parameter

File: src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx [hunk 16]
```diff
@@ -0,0 +1,32 @@
+'use client'
+import React, { useState } from 'react'
+import BatteryVoltage from '../animations/batteryVoltage.js'
+import ModalString from '../sharedComponents/ModalString';
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx:4]: ; at the end of import rows

File: src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+
+const VoltageCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.voltageCard);
+      const handleInputChange = (value: any, id: string) => {
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx:8]: delete id parameter

File: src/front_end/carsdata/app/components/sharedComponents/ModalString.tsx [hunk 18]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+
+const ModalString = (props: any) => {
+      const [input, setInput] = useState('');
+
+      const handleChange = (event: any) => {
+            props.handleInputChange(event.target.value, props.id);
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/sharedComponents/ModalString.tsx:7]: delete props.id parameter

**Review by Elisei-26 — CHANGES_REQUESTED** [review]

File: src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx:9]: delete the id parameter, it is not used, ad ; at the end of line 9

File: src/front_end/carsdata/app/components/batteryComponents/DeviceConsumption.tsx [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+
+const DeviceConsumptionCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.deviceConsumption);
+      const handleInputChange = (value: any) => {
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/DeviceConsumption.tsx:9]: ; at the end of line

File: src/front_end/carsdata/app/components/batteryComponents/FullyChargedCard.tsx [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+
+const FullyChargedCard = (props: any) => {
+      const [status, setstatus] = useState('off');
+      const change_status = () => {
+
+            if (status == 'on') {
+                  setstatus('off')
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/FullyChargedCard.tsx:10]: ; at the end of line

File: src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx:9]: delete the id parameter, it is not used, ad ; at the end of line 9

File: src/front_end/carsdata/app/components/batteryComponents/PercentageCard.tsx [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+
+const PercentageCard = (props: any) => {
+      const [inputValue, setInputValue] = useState(props.percentageBattery);
+      const handleInputChange = (value: any, id: string) => {
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/PercentageCard.tsx:9]: delete the id parameter, it is not used, ad ; at the end of line 9

File: src/front_end/carsdata/app/components/batteryComponents/RangeBatteryCard.tsx [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+
+
+const RangeBatteryCard = (props: any) => {
+      const [status, setstatus] = useState('off');
+      const change_status = () => {
+
+            if (status == 'on') {
+                  setstatus('off')
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/RangeBatteryCard.tsx:11]: ; at the end of line

File: src/front_end/carsdata/app/components/batteryComponents/SerialNumberCard.tsx [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/SerialNumberCard.tsx:9]: ; at the end of line

File: src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx:9]: delete the id parameter, it is not used, ad ; at the end of line 9

File: src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx [hunk 27]
```diff
@@ -0,0 +1,32 @@
...
+            setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx:9]: delete the id parameter, it is not used, ad ; at the end of line 9

**Review by Elisei-26 — CHANGES_REQUESTED** [review]

File: src/front_end/carsdata/app/components/animations/charging_time.js [hunk 28]
```diff
@@ -0,0 +1,15 @@
+import "./style.css"
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/animations/charging_time.js:1]: ; at the end of all styles imports

**Review by Elisei-26 — CHANGES_REQUESTED** [review]

File: src/front_end/carsdata/app/components/animations/rangeBattery.js [hunk 29]
```diff
@@ -1,6 +1,6 @@
 import "./style.css"
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/animations/rangeBattery.js:1]: add ; to all import rows in all files

File: src/front_end/carsdata/app/components/batteryComponents/BatteryStateCard.tsx [hunk 30]
```diff
@@ -1,11 +1,11 @@
 'use client'
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/BatteryStateCard.tsx:1]: ; here too

File: src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx [hunk 31]
```diff
@@ -1,11 +1,11 @@
...
-import Modal from '../sharedComponents/ModalString'
+import React, { useState } from 'react';
+import ChargingTime from '../animations/chargingTime.js';
+import Modal from '../sharedComponents/ModalString';
 
 const ChargingTimeCard = (props: any) => {
       const [inputValue, setInputValue] = useState(props.chargingTime);
-      const handleInputChange = (value: any, id: string) => {
+      const handleInputChange = (value: any) => {
             setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/Charging_time.tsx:9]: forgot ;

File: src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx [hunk 32]
```diff
@@ -1,11 +1,11 @@
...
-import Modal from '../sharedComponents/ModalString'
+import React, { useState } from 'react';
+import LifeCycle from '../animations/lifeCycle.js';
+import Modal from '../sharedComponents/ModalString';
 
 const LifeCycleCard = (props: any) => {
       const [inputValue, setInputValue] = useState(props.life_cycle);
-      const handleInputChange = (value: any, id: string) => {
+      const handleInputChange = (value: any) => {
             setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/LifeCycleCard.tsx:9]: forgot ;

File: src/front_end/carsdata/app/components/batteryComponents/PercentageCard.tsx [hunk 33]
```diff
@@ -1,11 +1,11 @@
...
-import ModalString from '../sharedComponents/ModalString'
+import React, { useState } from 'react';
+import BatteryPercentage from '../animations/batteryPercentage.js';
+import ModalString from '../sharedComponents/ModalString';
 
 const PercentageCard = (props: any) => {
       const [inputValue, setInputValue] = useState(props.percentageBattery);
-      const handleInputChange = (value: any, id: string) => {
+      const handleInputChange = (value: any) => {
             setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/PercentageCard.tsx:9]: forgot ;

File: src/front_end/carsdata/app/components/batteryComponents/RangeBatteryCard.tsx [hunk 34]
```diff
@@ -13,15 +13,15 @@ const RangeBatteryCard = (props: any) => {
                   setstatus('on');
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/RangeBatteryCard.tsx:13]: ; line 11

File: src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx [hunk 35]
```diff
@@ -1,11 +1,11 @@
...
-import Modal from '../sharedComponents/ModalString'
+import React, { useState } from 'react';
+import Temperature from '../animations/temperature.js';
+import Modal from '../sharedComponents/ModalString';
 
 const TemperatureCard = (props: any) => {
       const [inputValue, setInputValue] = useState(props.temperature);
-      const handleInputChange = (value: any, id: string) => {
+      const handleInputChange = (value: any) => {
             setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/TemperatureCard.tsx:9]: forgot ;

File: src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx [hunk 36]
```diff
@@ -1,11 +1,11 @@
...
-import BatteryVoltage from '../animations/batteryVoltage.js'
+import React, { useState } from 'react';
+import BatteryVoltage from '../animations/batteryVoltage.js';
 import ModalString from '../sharedComponents/ModalString';
 
 const VoltageCard = (props: any) => {
       const [inputValue, setInputValue] = useState(props.voltageCard);
-      const handleInputChange = (value: any, id: string) => {
+      const handleInputChange = (value: any) => {
             setInputValue(value)
```
Comment (Elisei-26) [inline src/front_end/carsdata/app/components/batteryComponents/VoltageCard.tsx:9]: forgot ;

**Review by Elisei-26 — CHANGES_REQUESTED** [review]

**Review by Elisei-26 — APPROVED** [review]

**Review by GeorgeCimpoeru — APPROVED** [review]

**Pull request merged**